### PR TITLE
IGP 26

### PR DESCRIPTION
**Pull request by thrilok209**

### CODE REVIEW

File: contracts/payloads/IGP26/PayloadIGP26.sol [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+        // Set user supply config for the vault on Liquidity Layer.
+        {
+            AdminModuleStructs.UserSupplyConfig[]
+                memory configs_ = new AdminModuleStructs.UserSupplyConfig[](1);
+
+            configs_[0] = getUserSupplyData(newConstants.supplyToken, oldVaultAddress);
+
+            (uint256 baseAllowance, ) = getAllowance(newConstants.supplyToken);
+
+            configs_[0].baseWithdrawalLimit = baseAllowance;
```
Comment (bergben) [inline contracts/payloads/IGP26/PayloadIGP26.sol:609]: we have a terrible UX at Liquidity AdminModule so base withdrawal limit input must be in raw, same for base debt ceiling and max debt ceiling :smiling_face_with_tear: I.e. currently the input is in normal token amount, but it must be `normalTokenAmount * 1e12 / exchangePrice`. 

we can get the exchange price for a token like this: 
```
uint256 exchangePriceAndConfig_ = LIQUIDITY.readFromStorage(
                LiquiditySlotsLink.calculateMappingStorageSlot(
                    LiquiditySlotsLink.LIQUIDITY_EXCHANGE_PRICES_MAPPING_SLOT,
                    token_
                )
            );
(uint256 supplyExchangePrice, uint256 borrowExchangePrice) = LiquidityCalcs
                .calcExchangePrices(exchangePriceAndConfig_);
```

Comment (thrilok209) [reply contracts/payloads/IGP26/PayloadIGP26.sol:609]: @bergben Before we use to set normal amount, right?

File: contracts/payloads/IGP26/PayloadIGP26.sol [hunk 2]
```diff
@@ -657,6 +707,22 @@ contract PayloadIGP26 {
...
+            address[] memory tokens = new address[](2);
+
+            {   
+                protocols[0] = newVaultAddress;
+                tokens[0] = newConstants.supplyToken;
+
+                protocols[1] = newVaultAddress;
+                tokens[1] = newConstants.borrowToken;
+
+                FLUID_RESERVE.approve(protocols, tokens, amounts);
```
Comment (bergben) [inline contracts/payloads/IGP26/PayloadIGP26.sol:723]: if the token is ETH, we need to skip setting any allowance for the token via reserveContract.approve(), as this is not implemented there (yet).

File: contracts/payloads/IGP26/PayloadIGP26.sol (hunk 1 above)
Comment (bergben) [reply contracts/payloads/IGP26/PayloadIGP26.sol]: Resolved. Github is not showing me a resolve button here idk why

**Review by bergben — APPROVED** [review]

**Pull request merged**